packages/openapi-generator/src/generator.ts

```diff
@@ -44,17 +49,26 @@ const logger = createLogger('openapi-generator');
  * @param options Options to configure generation.
  */
 export async function generate(options: GeneratorOptions): Promise<void> {
-  options.serviceMapping =
-    options.serviceMapping ||
-    resolve(options.input.toString(), 'service-mapping.json');
+  return generateWithParsedOptions(parseGeneratorOptions(options));
```

Comment:
I do not like that in the generate method there is something done parsing the options. This should take place in my opinion in the command (cli/index.ts). I found it also hard to understand what is the difference between the options and parsed options from the `ParsedGeneratorOptions` type.

Reply:
Actually this was done before already, only that it wasn't explicit.
I was also struggling with this, because I don't see a good way to unify the the commandline and programmatic use.
The cli parses the arguments (`this.parse`) and if the generation is invoked programatically, it should behave the same way.

Before, the default values where assigned/assumed everywhere in the code.

`GeneratorOptions` are the options that we allow to set when using the generator programatically. `ParsedGeneratorOptions` are the `GeneratorOptions`, only that default values and default parsing (e.g. `input => resolve(input)`) is already handled. You don't have to take care of optional values anymore.

Reply:
Leave as is, but document more thoroughly.

Reply:
check